prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,273 @@
+from dataclasses import dataclass
+from math import exp, log, pi, sqrt
+from statistics import geometric_mean
+
+from prereise.gather.griddata.transmission.const import (
+    epsilon_0,
+    mu_0,
+    relative_permeability,
+    resistivity,
+)
+
+
+@dataclass
+class Conductor:
+    """Represent a single conductor (which may be a stranded compsite).
+
+    :param float radius: outer radius of conductor.
+    :param str material: material of conductor. Used to calculate ``resistance_per_km``
+        and ``gmr`` if these aren't passed to the constructor, unnecessary otherwise.
+    :param float resistance_per_km: outer radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float gmr: geometric mean radius of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    :param float area: cross-sectional area of conductor. Will be estimated from
+        other parameters if it isn't passed.
+    """
+
+    radius: float
+    material: str = None
+    resistance_per_km: float = None
+    gmr: float = None
+    area: float = None
+
+    def __post_init__(self):
+        # Validate inputs
+        if self.gmr is None and (self.material is None or self.radius is None):
+            raise ValueError(
+                "If gmr is not provided, material and radius are needed to estimate"
+            )
+        if self.resistance_per_km is None and (
+            self.material is None or self.radius is None
+        ):
+            raise ValueError(
+                "If resistance_per_km is not provided, "
+                "material and radius are needed to estimate"
+            )
+
+        if self.gmr is None:
+            try:
+                self.permeability = relative_permeability[self.material]
```

Comment:
Not sure whether we would like these attributes which are assigned/calculated in post initialization to be shown once the data object is printed, if so, we could do `permeability: float = field(init=False) ` together with other class attributes.

Reply:
TIL! I've updated everything so that all these intermediately-calculated or looked-up values are contained in the `repr`:
```python
>>> from prereise.gather.griddata.transmission.geometry import Conductor
>>> c = Conductor(radius=0.1, material="aluminum")
>>> c
Conductor(radius=0.1, material='aluminum', resistance_per_km=0.0008435211983870453, gmr=0.12840254166877416, area=0.031415926535897934, permeability=1)
```